src/frontend/metrics/generateMetricsReport.ts

```diff
@@ -0,0 +1,38 @@
+import type {ReadonlyDeep} from 'type-fest';
+import type {Observation} from '@mapeo/schema';
+import positionToCountries from './positionToCountries';
+
+export default function generateMetricsReport({
```

Comment:
This function takes a bunch of dependencies. It's more verbose, but I think it has some advantages:

- Easier to port to Electron in the future.
- Clearer to the caller what goes in; data isn't pulled from anywhere unexpected.
- Easier to test.